### PR TITLE
Add record Deconstruct section

### LINKED ISSUE #22583: Please add note on record-class deconstruct here
Since this page is all about Deconstruct, there should at least be a note that with C# 9 record-classes come with a built-in deconstruct function. Probably add a short section on using that.

edit by @billwagner. Add blank line to unbold description.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ec91dfa8-265f-a220-fe14-14047375ee93
* Version Independent ID: b2afe72f-5b30-8938-7b64-9f45056df36a
* Content: [Deconstructing tuples and other types](https://docs.microsoft.com/en-us/dotnet/csharp/deconstruct)
* Content Source: [docs/csharp/deconstruct.md](https://github.com/dotnet/docs/blob/master/docs/csharp/deconstruct.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #22583

### CODE REVIEW

File: docs/csharp/deconstruct.md [hunk 1]
```diff
@@ -60,7 +59,7 @@ The following example illustrates the use of tuples with discards. The `QueryCit
 
 ## Deconstructing user-defined types
 
-C# does not offer built-in support for deconstructing non-tuple types. However, as the author of a class, a struct, or an interface, you can allow instances of the type to be deconstructed by implementing one or more `Deconstruct` methods. The method returns void, and each value to be deconstructed is indicated by an [out](language-reference/keywords/out-parameter-modifier.md) parameter in the method signature. For example, the following `Deconstruct` method of a `Person` class returns the first, middle, and last name:
+C# does not offer built-in support for deconstructing non-tuple types other than [`record` types](#deconstructing-a-record-type). However, as the author of a class, a struct, or an interface, you can allow instances of the type to be deconstructed by implementing one or more `Deconstruct` methods. The method returns void, and each value to be deconstructed is indicated by an [out](language-reference/keywords/out-parameter-modifier.md) parameter in the method signature. For example, the following `Deconstruct` method of a `Person` class returns the first, middle, and last name:
```
Comment (IEvangelist) [inline docs/csharp/deconstruct.md:62]: The only other built-in support is for [`KeyValuePair<TKey, TValue>`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Collections/DictionaryEntry.cs#L38-L42):

```csharp
foreach (var (key, value) in new Dictionary<int, string>
         {
             [1] = "One", [2] = "Two", [3] = "Three"
         })
{
    Console.WriteLine($"{key} = {value}");
}
```

I knew about the KVP, but did a [quick search for other options](https://github.com/dotnet/runtime/search?q=Deconstruct).

Comment (tdykstra) [inline docs/csharp/deconstruct.md:62]:
```suggestion
C# does not offer built-in support for deconstructing non-tuple types other than [`record`](#deconstructing-a-record-type) and [DictionaryEntry](xref:System.Collections.DictionaryEntry.Deconstruct). However, as the author of a class, a struct, or an interface, you can allow instances of the type to be deconstructed by implementing one or more `Deconstruct` methods. The method returns void, and each value to be deconstructed is indicated by an [out](language-reference/keywords/out-parameter-modifier.md) parameter in the method signature. For example, the following `Deconstruct` method of a `Person` class returns the first, middle, and last name:
```

File: docs/csharp/deconstruct.md [hunk 2]
```diff
@@ -60,7 +59,7 @@ The following example illustrates the use of tuples with discards. The `QueryCit
...
 
-C# does not offer built-in support for deconstructing non-tuple types. However, as the author of a class, a struct, or an interface, you can allow instances of the type to be deconstructed by implementing one or more `Deconstruct` methods. The method returns void, and each value to be deconstructed is indicated by an [out](language-reference/keywords/out-parameter-modifier.md) parameter in the method signature. For example, the following `Deconstruct` method of a `Person` class returns the first, middle, and last name:
+C# does not offer built-in support for deconstructing non-tuple types other than [`record`](#deconstructing-a-record-type) and [DictionaryEntry](xref:System.Collections.DictionaryEntry.Deconstruct). However, as the author of a class, a struct, or an interface, you can allow instances of the type to be deconstructed by implementing one or more `Deconstruct` methods. The method returns void, and each value to be deconstructed is indicated by an [out](language-reference/keywords/out-parameter-modifier.md) parameter in the method signature. For example, the following `Deconstruct` method of a `Person` class returns the first, middle, and last name:
```
Comment (tdykstra) [inline docs/csharp/deconstruct.md:62]:
```suggestion
C# does not offer built-in support for deconstructing non-tuple types other than [`record`](#deconstructing-a-record-type) and [DictionaryEntry](xref:System.Collections.DictionaryEntry.Deconstruct%2A). However, as the author of a class, a struct, or an interface, you can allow instances of the type to be deconstructed by implementing one or more `Deconstruct` methods. The method returns void, and each value to be deconstructed is indicated by an [out](language-reference/keywords/out-parameter-modifier.md) parameter in the method signature. For example, the following `Deconstruct` method of a `Person` class returns the first, middle, and last name:
```

File: docs/csharp/deconstruct.md [hunk 3]
```diff
@@ -60,7 +59,7 @@ The following example illustrates the use of tuples with discards. The `QueryCit
...
 
-C# does not offer built-in support for deconstructing non-tuple types. However, as the author of a class, a struct, or an interface, you can allow instances of the type to be deconstructed by implementing one or more `Deconstruct` methods. The method returns void, and each value to be deconstructed is indicated by an [out](language-reference/keywords/out-parameter-modifier.md) parameter in the method signature. For example, the following `Deconstruct` method of a `Person` class returns the first, middle, and last name:
+C# does not offer built-in support for deconstructing non-tuple types other than [`record`](#deconstructing-a-record-type) and [DictionaryEntry](xref:System.Collections.DictionaryEntry.Deconstruct%2A). However, as the author of a class, a struct, or an interface, you can allow instances of the type to be deconstructed by implementing one or more `Deconstruct` methods. The method returns void, and each value to be deconstructed is indicated by an [out](language-reference/keywords/out-parameter-modifier.md) parameter in the method signature. For example, the following `Deconstruct` method of a `Person` class returns the first, middle, and last name:
```
Comment (tdykstra) [inline docs/csharp/deconstruct.md:62]:
```suggestion
C# does not offer built-in support for deconstructing non-tuple types other than the [`record`](#deconstructing-a-record-type) and [DictionaryEntry](xref:System.Collections.DictionaryEntry.Deconstruct%2A) types. However, as the author of a class, a struct, or an interface, you can allow instances of the type to be deconstructed by implementing one or more `Deconstruct` methods. The method returns void, and each value to be deconstructed is indicated by an [out](language-reference/keywords/out-parameter-modifier.md) parameter in the method signature. For example, the following `Deconstruct` method of a `Person` class returns the first, middle, and last name:
```

File: docs/csharp/deconstruct.md (hunk 1 above)
Comment (tdykstra) [reply docs/csharp/deconstruct.md]: Good catch, thanks!

**Review by IEvangelist — APPROVED** [review]
Looks good to me, :shipit: when you're ready.

**Pull request merged**